### DISCUSSION TITLE
error when installing

### DISCUSSION THREAD

**Discussion by ghost** [spatie/laravel-backup#1051] [Q&A] [answered]
$ composer require spatie/laravel-backup
Using version ^6.8 for spatie/laravel-backup
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 2 installs, 0 updates, 0 removals
  - Installing spatie/db-dumper (2.15.3): Loading from cache
  - Installing spatie/laravel-backup (6.8.1): Loading from cache
spatie/laravel-backup suggests installing laravel/slack-notification-channel (Required for sending notifications via Sla
ck)
Writing lock file
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover --ansi

   Illuminate\Contracts\Container\BindingResolutionException

  Target [Spatie\Backup\Tasks\Cleanup\CleanupStrategy] is not instantiable while building [Spatie\Backup\Commands\Cleanu
pCommand].

  at C:\xampp\htdocs\laravel\vendor\laravel\framework\src\Illuminate\Container\Container.php:982
    978|         } else {
    979|             $message = "Target [$concrete] is not instantiable.";
    980|         }
    981|
  > 982|         throw new BindingResolutionException($message);
    983|     }
    984|
    985|     /**
    986|      * Throw an exception for an unresolvable primitive.

  1   C:\xampp\htdocs\laravel\vendor\laravel\framework\src\Illuminate\Container\Container.php:816
      Illuminate\Container\Container::notInstantiable("Spatie\Backup\Tasks\Cleanup\CleanupStrategy")

  2   C:\xampp\htdocs\laravel\vendor\laravel\framework\src\Illuminate\Container\Container.php:261
      Illuminate\Container\Container::build("Spatie\Backup\Tasks\Cleanup\CleanupStrategy")
Script @php artisan package:discover --ansi handling the post-autoload-dump event returned with error code 1

Installation failed, reverting ./composer.json to its original content.

**Comment by j5k** [comment]
Same here with newest Homestead and updated composer packages.

**Comment by j5k** [comment, marked as answer]
I fixed it with the following commands:
`rm -rf vendor/* \n`
`git clean -ifx` (deletes some caches, etc.)
`composer update`
`composer require spatie/laravel-backup`

Maybe an artisan cache cleaning is the key hint
